py/server/tests/test_iceberg.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+#
+
+from deephaven import dtypes
+from deephaven.column import Column, ColumnType
+
+from tests.testbase import BaseTestCase
+from deephaven.experimental import s3, iceberg
+
+
+class IcebergTestCase(BaseTestCase):
+    """ Test cases for the deephaven.iceberg module (performed locally) """
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def test_instruction_create_empty(self):
+        iceberg_instructions = iceberg.IcebergInstructions()
+
+    def test_instruction_create_with_s3_instructions(self):
+        s3_instructions = s3.S3Instructions(region_name="us-east-1",
+                                            access_key_id="some_access_key_id",
+                                            secret_access_key="som_secret_access_key"
+                                            )
+        iceberg_instructions = iceberg.IcebergInstructions(data_instructions=s3_instructions)
+
+    def test_instruction_create_with_col_renames(self):
+        renames = {
+            "old_name_a": "new_name_a",
+            "old_name_b": "new_name_b",
+            "old_name_c": "new_name_c"
+        }
+        iceberg_instructions = iceberg.IcebergInstructions(column_renames=renames)
+
+    def test_instruction_create_with_table_definitition_dict(self):
+        table_def={
+            "x": dtypes.int32,
+            "y": dtypes.double,
+            "z": dtypes.double,
+        }
+
+        iceberg_instructions = iceberg.IcebergInstructions(table_definition=table_def)
+
+    def test_instruction_create_with_table_definitition_list(self):
+        table_def=[
+            Column(
+                "Partition", dtypes.int32, column_type=ColumnType.PARTITIONING
+            ),
+            Column("x", dtypes.int32),
+            Column("y", dtypes.double),
+            Column("z", dtypes.double),
+        ]
+
+        iceberg_instructions = iceberg.IcebergInstructions(table_definition=table_def)
```

Comment:
Missing tests:
* `IcebergCatalogAdapter`
* `s3_rest_adapter`
* `aws_glue_adapter`

Reply:
These objects aren't instantiable without a working S3 or AWS backend.  I have tested manually, but not sure how or if we can test these in CI.

Reply:
`IcebergCatalogAdapter` is tested as a Java object in CI (`IcebergToolsTest.java`)

Reply:
1. Testing the Java is not the same as making sure the python actually works.  Python tests don't need to be the same detail, but they should confirm that the code runs.
2. In python, I think we test against Kafka, SQL, etc.  Is there a reason that we shouldn't actually be testing against some S3 here?  Is it significantly harder to setup than these other cases?
3. There is no way to run the tests in a repeatable way when there is S3 available if the tests don't exist.
4. I have low confidence that this (or any other code) will keep working without some kind of test.

Reply:
After discussing with @devinrsmith and @malhotrashivam , it is significantly harder to test the python wrappers directly.  But these wrappers are very light wrapping over better tested Java objects.

Reply:
Regardless, the python isn't tested.

Reply:
We're quite confident that we don't want build additional dockerized integration tests. They tend to be flaky, and the work involved greatly exceeds the scope of this ticket. We'll file a ticket to expand testcontainers-based testing for glue, but per @devinrsmith it appears to be a paid feature so we may not want to add it.

I think manual testing is enough for such a thin wrapper, and I don't intend to sponsor docker-based automation at this time.

Reply:
Created the following ticket to address missing CI testing: https://github.com/deephaven/deephaven-core/issues/5656